tests/test_enterprise/api/test_views.py

```diff
@@ -627,20 +614,175 @@ def test_enterprise_customer_courses(
 
         assert response == expected
 
+    @ddt.data(
+        (
+            ENTERPRISE_CUSTOMER_CATALOG_DETAIL_ENDPOINT,
+            False, False,
+            '',
+            {
+                'detail': "User must be a staff user or "
+                          "associated with the endpoint's appointed common Enterprise Customer."
+            },
+        ),
+        (
+            ENTERPRISE_CUSTOMER_CATALOG_DETAIL_ENDPOINT,
+            False, True,
+            '',
+            {
+                'detail': "User must be a staff user or "
+                          "associated with the endpoint's appointed common Enterprise Customer."
+            },
+        ),
+        (
+            ENTERPRISE_CUSTOMER_CATALOG_DETAIL_ENDPOINT,
+            True, False,  # Is a staff user, so it's allowed.
+            'no catalog for query so use empty string, i.e. get everything',
+            fake_catalog_api.FAKE_SEARCH_ALL_RESULTS,
+        ),
+        (
+            ENTERPRISE_CUSTOMER_CATALOG_DETAIL_ENDPOINT,
+            True, True,
+            '',
+            fake_catalog_api.FAKE_SEARCH_ALL_RESULTS,
+        ),
+        (
+            ENTERPRISE_CUSTOMER_CATALOG_DETAIL_ENDPOINT,
+            True, False,
+            'no catalog for query so use empty string, i.e. get everything',
+            fake_catalog_api.FAKE_SEARCH_ALL_RESULTS,
+        ),
+        (
+            ENTERPRISE_CUSTOMER_CATALOG_DETAIL_ENDPOINT,
+            True, True,
+            'edX Demonstration Course',
+            {
+                'count': 2,
+                'next': None,
+                'previous': None,
+                'results': [
+                    fake_catalog_api.FAKE_SEARCH_ALL_COURSE_RESULT,
+                    fake_catalog_api.FAKE_SEARCH_ALL_SHORT_COURSE_RESULT,
+                ],
+            },
+        ),
+        (
+            PROGRAMS_DETAIL_ENDPOINT,
```

Comment:
Yes but similarly, one wouldn't expect the Programs client code to be under the `CourseCatalogApiClient` class (at least not me!), but it is. In the last PR, I had a significant refactor of this class into separate parts -- a class for targeting the Programs endpoint, one for the catalogs endpoint, etc. -- but Doug told to hold back on this, so I didn't bring that into this PR and decided to bite the nail and continue the current umbrella-ing of the Programs endpoint under anything "catalog".